### PR TITLE
Release 3.0.0

### PR DESCRIPTION
**Pull request by MrBurtyyy**
### Use of `ServiceLocator`

- The use of the `ServiceLocator` class has been dropped in order to simplify the development process of the SDK.
- The `yoti-sdk-api` and `yoti-sdk-impl` modules have been merged, with the `yoti-sdk-impl` module then being removed.

### Builders

All builders have now been in-lined into their respective classes, and the use of factories has been dropped (they are no longer neccessary due to the removal of the use of `ServiceLocator`).  Example below:

```java
YotiClient client = YotiClientBuilder.newInstance().build() // OLD

YotiClient client = YotiClient.builder().build() // NEW
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,14 +100,14 @@ If you are using Maven, you need to add the following dependency:
 ```xml
 <dependency>
     <groupId>com.yoti</groupId>
-    <artifactId>yoti-sdk-impl</artifactId>
-    <version>2.12.0</version>
+    <artifactId>yoti-sdk-api</artifactId>
+    <version>3.0.0-SNAPSHOT</version>
```
Comment (echarrod) [inline README.md:104]:
```suggestion
    <version>3.0.0</version>
```

File: README.md [hunk 2]
```diff
@@ -100,14 +100,14 @@ If you are using Maven, you need to add the following dependency:
...
 </dependency>
 ```
 
 If you are using Gradle, here is the dependency to add:
 
-`compile group: 'com.yoti', name: 'yoti-sdk-impl', version: '2.12.0'`
+`compile group: 'com.yoti', name: 'yoti-sdk-api', version: '3.0.0-SNAPSHOT'`
```
Comment (echarrod) [inline README.md:110]:
```suggestion
`compile group: 'com.yoti', name: 'yoti-sdk-api', version: '3.0.0'`
```

**Review by echarrod — CHANGES_REQUESTED** [review]
We need to update the versions from `3.0.0-SNAPSHOT` to `3.0.0` don't we?

**Comment by MrBurtyyy** [comment]
> We need to update the versions from `3.0.0-SNAPSHOT` to `3.0.0` don't we?

These will be done just before release.  The reason we don't do it straight away is so QA can easily run the demo projects without having to manually change the versions back to `-SNAPSHOT` versions

File: yoti-sdk-api/src/main/java/com/yoti/api/client/docs/DocScanClient.java [hunk 3]
```diff
@@ -59,14 +108,59 @@
...
+            LOG.debug("Loading key pair from '{}'", kpSource);
+            return kpSource.getFromStream(new KeyStreamVisitor());
+        } catch (IOException e) {
+            throw new InitialisationException("Cannot load key pair", e);
+        }
+    }
+
+    public static class Builder {
+
+        // FIXME: Should this really be a static property?????
```
Comment (echarrod) [inline yoti-sdk-api/src/main/java/com/yoti/api/client/docs/DocScanClient.java:138]:
```suggestion
```

**Review by bucky-boy — APPROVED** [review]
looks good to me.

**Review by jlmartyres — APPROVED** [review]

**Review by echarrod — APPROVED** [review]
:rocket:

**Pull request merged**